.github/workflows/scripts/send-notification-on-change.js

```diff
@@ -0,0 +1,139 @@
+  const clientSecret = process.env.OAUTH2_CLIENT_SECRET;
+  const refreshToken = process.env.OAUTH2_REFRESH_TOKEN;
+
+  // validate params
+  if (!repo || !prNumber || !token || !clientId || !clientSecret || !refreshToken) {
+    console.error('Missing required environment variables.');
+    process.exit(1);
+  }
+
+  // the whole process is in a big try/catch. e.g. if the config file doesn't exist, github is down, etc.
+  try {
+    // Read and process the configuration file
+    const configFileContent = fs.readFileSync(configFilePath, 'utf-8');
+    const configRules = configFileContent
+      .split('\n')
+      .filter(line => line.trim() !== '' && !line.trim().startsWith('#')) // Ignore empty lines and comments
+      .map(line => {
+        const [regex, email] = line.split(':').map(part => part.trim());
+        return { regex: new RegExp(regex), email };
+      });
+
+    // Fetch changed files from github
+    const [owner, repoName] = repo.split('/');
+    const apiUrl = `https://api.github.com/repos/${owner}/${repoName}/pulls/${prNumber}/files`;
+    const response = await axios.get(apiUrl, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
+
+    const changedFiles = response.data.map(file => file.filename);
+    console.log('Changed files:', changedFiles);
+
+    // match file pathnames that are in the config and group them by email address
+    const matchesByEmail = {};
+    changedFiles.forEach(file => {
+      configRules.forEach(rule => {
+        if (rule.regex.test(file)) {
+          if (!matchesByEmail[rule.email]) {
+            matchesByEmail[rule.email] = [];
+          }
+          matchesByEmail[rule.email].push(file);
+        }
+      });
+    });
+
+    // Exit successfully if no matches were found
+    if (Object.keys(matchesByEmail).length === 0) {
+      console.log('No matches found. Exiting successfully.');
+      process.exit(0);
+    }
+
+    console.log('Grouped matches by email:', matchesByEmail);
+
+    // get ready to email the changes
+    const accessToken = await getAccessToken(clientId, clientSecret, refreshToken);
+
+    // Configure Nodemailer with OAuth2
+    //  service: 'Gmail',
+    const transporter = nodemailer.createTransport({
+      host: "smtp.gmail.com",
+      port: 465,
+      secure: true,
+      auth: {
+        type: 'OAuth2',
+        user: 'info@prebid.org',
+        clientId: clientId,
+        clientSecret: clientSecret,
+        refreshToken: refreshToken,
+        accessToken: accessToken
+      },
+    });
+
+    // Send one email per recipient
+    for (const [email, files] of Object.entries(matchesByEmail)) {
+      const emailBody = `
+        ${email},
+        <p>
+        Files owned by you have been changed in open source ${repo}. The <a href="https://github.com/${repo}/pull/${prNumber}">pull request is #${prNumber}</a>. These are the files you own that have been modified:
```

Comment:
Let's change the word owned here. How about files relevant to your integration

Reply:
done